funcx_sdk/funcx/sdk/login_manager/manager.py

```diff
@@ -197,3 +197,15 @@ def get_funcx_web_client(
             app_name=app_name,
             authorizer=self._get_authorizer(FuncxScopes.resource_server),
         )
+
+    def validate_grant(self, web_client: FuncxWebClient) -> FuncxWebClient:
+        """Check the validity of the current auth grant (i.e., refresh token)
+        by sending a request to the API. If invalid, run a new auth flow and
+        return a new web client.
+        """
+        try:
+            web_client.get_version()
+        except globus_sdk.AuthAPIError:
```

Comment:
Will this create an AuthAPIError?  `curl` suggest that it would not:
```sh
$ curl --stderr - -sv4 https://api2.funcx.org/v2/version | grep -E '^[^*}{]'
> GET /v2/version HTTP/2
> Host: api2.funcx.org
> user-agent: curl/7.81.0
> accept: */*
> 
< HTTP/2 200 
< date: Thu, 15 Dec 2022 17:25:53 GMT
< content-type: application/json
< content-length: 8
< 
"1.0.6"
```
That is that with no authorization is needed for that endpoint, so it nominally doesn't validate the token, if sent.

Reply:
This does raise an `AuthAPIError` because `FuncXWebClient` inherits it's `get()` method from `globus_sdk.BaseClient`, which calls the OAuth2 token endpoint.